### PR TITLE
Add checks

### PR DESCRIPTION
**Pull request by lewismarshall**
Need to fail early so this PR aims to add pre-flight checks to restore and save

### CODE REVIEW

**Comment by lewismarshall** [comment]
Fixes #22

**Comment by lewismarshall** [comment]
Fixes #11

**Comment by vinn946** [comment]
Bug with `--archive-dir` flag, which points to an incorrect directory:
```
[vincentlam@td-dsp01 crown-high-elk]$ ./downloads/artefactor publish --archive-dir ~/crown-high-elk/downloads/ --docker-registry registry.dsp.internal
Loading image from /home/vincentlam/crown-high-elk/downloads/home/vincentlam/crown-high-elk/downloads/minio~minio~~RELEASE.2018-04-12T23-41-09Z.docker.tar.
```
Problem persists even when using default option:
```
[vincentlam@td-dsp01 crown-high-elk]$ pwd
/home/vincentlam/crown-high-elk
[vincentlam@td-dsp01 crown-high-elk]$ ./downloads/artefactor publish --docker-registry registry.dsp.internal
Loading image from downloads/downloads/minio~minio~~RELEASE.2018-04-12T23-41-09Z.docker.tar.
Error: load image problem for downloads/downloads/minio~minio~~RELEASE.2018-04-12T23-41-09Z.docker.tar:stat downloads/downloads/minio~minio~~RELEASE.2018-04-12T23-41-09Z.docker.tar: no such file or directory
```

File: pkg/docker/docker.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func GetImages(path string, registry string) ([]Image, error) {
 				return nil, err
 			}
 			images = append(images, Image{
-				FileName:     filepath.Join(path, file),
+				FileName:     file,
```
Comment (lewismarshall) [inline pkg/docker/docker.go]: @vinn946 here's the fix (included in squash)

**Comment by lewismarshall** [comment]
@vinn946 Fixed and backed as https://github.com/appvia/artefactor/releases/tag/v0.2.0-rc3

**Pull request merged**